test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs

```diff
@@ -80,7 +81,36 @@ public class ConnectionFilterTests
                         "Hello World!");
                 }
             }
-       }
+        }
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Test hangs after execution on Mono.")]
+        public async Task ThrowingSynchronousConnectionFilterDoesNotCrashServer()
+        {
+            var serviceContext = new TestServiceContext()
+            {
+                ConnectionFilter = new ThrowingConnectionFilter()
+            };
+
+            using (var server = new TestServer(App, serviceContext))
+            {
+                using (var connection = new TestConnection())
+                {
+                    try
+                    {
+                        await connection.SendEnd(
+                            "POST / HTTP/1.0",
+                            "",
+                            "Hello World?");
+                    }
+                    catch (IOException)
+                    {
+                        // Will throw because the exception in the connection filter will close the connection.
+                        Assert.True(true);
```

Comment:
Did this test fail before your change? If I remember correctly, the entire server would crash in the scenario previously, so I would expect an exception from SendEnd before this change.

Maybe before TestServer.Dispose was throwing?